sys/posix/sockets/posix_sockets.c

```diff
@@ -479,6 +498,16 @@ int socket(int domain, int type, int protocol)
             errno = EAFNOSUPPORT;
             res = -1;
     }
+
+    char *buf_ptr = malloc(PEEK_BUF_SIZE * sizeof(char));
```

Comment:
Not sure why `malloc` is used here, rather than `recv_buf`. The socket pools here should make clear, that this module tries to avoid using `malloc`, which should always only be a fall-back if nothing else is possible in embedded programming.